crates/turbopack-ecmascript-runtime/js/src/shared/runtime-utils.ts

```diff
@@ -197,3 +232,172 @@ function requireContext(
 function getChunkPath(chunkData: ChunkData): ChunkPath {
   return typeof chunkData === "string" ? chunkData : chunkData.path;
 }
+
+function isAsyncExports(
+  exports: Exports | AsyncModulePromise
+): exports is AsyncModulePromise {
+  return typeof exports.then === "function";
+}
+
+function isPromise<T = any>(maybePromise: any): maybePromise is Promise<T> {
+  return (
+    maybePromise != null &&
+    typeof maybePromise === "object" &&
+    "then" in maybePromise &&
+    typeof maybePromise.then === "function"
+  );
+}
+
+function createPromise<T>() {
+  let resolve: (value: T | PromiseLike<T>) => void;
+  let reject: (reason?: any) => void;
+
+  const promise = new Promise<T>((res, rej) => {
+    reject = rej;
+    resolve = res;
+  });
+
+  return {
+    promise,
+    resolve: resolve!,
+    reject: reject!,
+  };
+}
+
+// everything below is just stolen from webpack
+const turbopackQueues = Symbol("turbopack queues");
```

Comment:
I wanted to use a const string as the alternative, but typescript doesn't like it (TS1170)
```ts
const turbopackQueues = typeof Symbol !== "undefined" ? Symbol("turbopack queues") : "__turbopack_queues__" as const;
```

[Typescript Playground](https://www.typescriptlang.org/play?#code/MYewdgzgLgBFCuAnARiADgQ2AawIrwFNCIYBeOATzQJADMYBlCgW1QBsYBCU8gInjAATArQCWYAoN4wA-IxbsAFLwQp0WbDACOhYrwCUMAFwxeAfTOrUmHGZ1ECEC9IwlQkKAG4AUAHpfMIEwAHoy3t5QVAQwAIIQFGDA+A4AYmBkMIqKhqQAfDAAbiCigoYAZDAA3tq6BADCIAJQJmDwrASIMAC+PpHUsfGJyYQZcQlJtWkA2gC6MBXViI4gbAWSJqgrBBjpPeF90WOJALIggvBsBACiAB6w5JXegVNW6jjDjjMmirRg3-aEExHCYOHL5IolMGFYqCHx7IA)